bundles/framework/timeseries/WMSAnimator.js

```diff
@@ -0,0 +1,180 @@
+        this._mapModule = sandbox.findRegisteredModuleInstance('MainMapModule');
+        this._sandbox = sandbox;
+        this._layer = this._sandbox.findMapLayerFromSelectedMapLayers(layerId);
+
+        var times = this.getTimes();
+        this._currentTime = times[times.length - 1];
+        this._subsetRange = [times[0], times[times.length - 1]];
+
+        this._doneCallback = null;
+        this._isBuffering = false;
+        this._isLoading = false;
+
+        this._sandbox.register(this);
+        var p;
+        for (p in this.__eventHandlers) {
+            if (this.__eventHandlers.hasOwnProperty(p)) {
+                sandbox.registerForEventByName(this, p);
+            }
+        }
+    }, {
+        __name: 'WMSAnimator',
+        getName: function () {
+            return this.__name;
+        },
+        _clazz: 'Oskari.mapframework.bundle.timeseries.WMSAnimator',
+        __eventHandlers: {
+            'ProgressEvent': function (event) {
+                if (event.getStatus() && this._layer.getId() === event.getId()) {
+                    this._isLoading = false;
+                    this._resolveWait();
+                }
+            }
+        },
+        onEvent: function (event) {
+            var handler = this.__eventHandlers[event.getName()];
+            if (!handler) {
+                return;
+            }
+            return handler.apply(this, [event]);
+        },
+        /**
+         * @method getTimes
+         * Returns all available time instants that can be shown
+         * @return {String[]} list of available timeseries times
+         */
+        getTimes: function () {
+            var times = this._layer.getAttributes().times;
+            if (!Array.isArray(times)) {
+                var interval = moment.duration(times.interval);
+                var end = moment(times.end);
+                var t = moment(times.start);
+                times = [t.toISOString()];
+                while (t.add(interval) < end) {
+                    times.push(t.toISOString());
+                }
+                times.push(end.toISOString());
+            }
+            return times;
+        },
+        init: function () { },
+        /**
+         * @method getCurrentTime
+         * Returns current selected time instant
+         * @return {String} current time as ISO-string
+         */
+        getCurrentTime: function () {
+            return this._currentTime;
+        },
+        /**
+         * @method getSubsetRange
+         * Returns current selected time range subset
+         * @return {String[]} range Array with 2 elements, start & end, ISO-string
+         */
+        getSubsetRange: function () {
+            return this._subsetRange.slice();
+        },
+        /**
+         * @method getSubsetRange
+         * Set current selected time range subset
+         * @param {String[]} range Array with 2 elements, start & end, ISO-string
+         */
+        setSubsetRange: function (range) {
```

Comment:
Should this check that the parameter is an array with 2 elements?

Reply:
Good question. Do we generally want to check that arguments are of correct type, or do we trust that the developer has read the docs?

Reply:
”It’s complicated” :) With simple values usually no checking is required, but with complicated values/object I’d prefer some validation. I guess this falls somewhere in the middle. I think the question can be answered by asking ”how difficult is it to find the root cause if someone passes an invalid value causing an error later on in the code”. That being said, its not mandatory to have a check here.